receiver/redisreceiver/redis_scraper.go

```diff
@@ -93,7 +95,11 @@ func (rs *redisScraper) Scrape(context.Context) (pmetric.Metrics, error) {
 	rs.recordKeyspaceMetrics(now, inf)
 	rs.recordRoleMetrics(now, inf)
 	rs.recordCmdStatsMetrics(now, inf)
-	return rs.mb.Emit(metadata.WithRedisVersion(rs.getRedisVersion(inf))), nil
+	var resourceMetricOptions []metadata.ResourceMetricsOption
+	if rs.cfg.MetricsBuilderConfig.ResourceAttributes.RedisVersion.Enabled {
+		resourceMetricOptions = append(resourceMetricOptions, metadata.WithRedisVersion(rs.getRedisVersion(inf)))
+	}
```

Comment:
I don't believe this is needed. Its a responsibility of the metrics builder and should be handled by `WithRedisVersion`. If it's not working for any reason, let's fix the metrics builder